### PR TITLE
doc: align dev checklist to applied practices in ORAS 1.1.0-rc.1 release

### LINKED ISSUE #213: Update release check list for ORAS CLI
To echo https://github.com/oras-project/oras/issues/1013, the steps for cutting a release branch in [release check list](https://oras.land/docs/community/developer_guide#release-checklist) should be updated to include below details:
- Preparations for a release PR
- How to merge a release PR and tag a release

### LINKED ISSUE #215: Update the documentation version from v1.0.0 to v1.1.0-rc.1
### Code of Conduct

- [X] I have read and agree to the ORAS' [Code of Conduct](https://github.com/oras-project/oras-www/blob/main/CODE_OF_CONDUCT.md)

### What part of document/web-page on oras.land is affected?

- [ ] The [installation guide](https://oras.land/docs/how_to_guides/installation/) is still v1.0.0. It needs to be changed to v1.1.0-rc.1
- [ ] Homebrew, Winget, Nix, and Snap only support ORAS v1.0.0 so far. The [installation guide](https://oras.land/docs/how_to_guides/installation/) should remind users that v1.1.0-rc.1 are not available on those package managers. They are still WIP or to be upgraded.
- [ ] There is a version number on the website. However, it's not appropriate to state that the whole ORAS project is using the same version as the ORAS CLI (e.g. v1.0.0). The ORAS libraries like ORAS-go, ORAS-py, and ORAS-dotnet have different versioning strategies, Should the version number need to be removed from the ORAS website or we state that this version number is the CLI version? 
image

### Additional comments?

_No response_

### PR DESCRIPTION
**Pull request by qweeah**
Resolves #213

### CODE REVIEW

File: docs/community/developer_guide.mdx [hunk 1]
```diff
@@ -83,14 +83,19 @@ To update or add new dependencies, run `go get <package name>`.
...
-1. Make fresh clone of the repo after all above steps are completed. Create a new tag for the version prefixed with "v" and push the tag directly to the repo.
+1. Open a PR to update Github Actions with latest version of golang: replace go version of [binary](https://github.com/oras-project/oras/blob/main/.github/workflows/release-github.yml#L32) and [image](https://github.com/oras-project/oras/blob/main/Dockerfile#L14) to latest stable one
+
+### Call for vote
+1. Open a release PR to update oras version: replace [current stable version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
+1. Send a message to slack channel to call for vote on releasing. If the vote passes, merge the PR.
+ - The target commit should be the SHA digest of the last commit in the release PR
+ - Make sure that the PR is merged with `Create a merge commit` option
+ - (optional) cut off a release branch named `release-<major>.<minor>` if there is not one already
+### Release
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:97]: nit: space before the title

File: docs/community/developer_guide.mdx [hunk 2]
```diff
@@ -83,14 +83,19 @@ To update or add new dependencies, run `go get <package name>`.
...
-   - Update oras version: replace [current stable version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
-1. After the release PR got merged, [create an issue](https://github.com/oras-project/oras/issues/new) to call for vote on cutting off a release branch named `release-<major>.<minor>` based on the version update commit.
-1. Make fresh clone of the repo after all above steps are completed. Create a new tag for the version prefixed with "v" and push the tag directly to the repo.
+1. Open a PR to update Github Actions with latest version of golang: replace go version of [binary](https://github.com/oras-project/oras/blob/main/.github/workflows/release-github.yml#L32) and [image](https://github.com/oras-project/oras/blob/main/Dockerfile#L14) to latest stable one
+
+### Call for vote
+1. Open a release PR to update oras version: replace [current stable version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:92]: What's the target branch for this PR?

File: docs/community/developer_guide.mdx [hunk 3]
```diff
@@ -83,14 +83,19 @@ To update or add new dependencies, run `go get <package name>`.
...
+1. Send a message to slack channel to call for vote on releasing. If the vote passes, merge the PR.
+ - The target commit should be the SHA digest of the last commit in the release PR
+ - Make sure that the PR is merged with `Create a merge commit` option
+ - (optional) cut off a release branch named `release-<major>.<minor>` if there is not one already
+### Release
+1. Make fresh clone of the repo after all above steps are completed. Create a new tag for the version prefixed with "v" and push the tag directly to the repo. The tag should be created on the last commit of release PR, **NOT** the merge commit.
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:98]: The tag should be created on the **voted** commit

File: docs/community/developer_guide.mdx (hunk 2 above)
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:92]: Should reference the version bump up commit in the PR description for vote.

File: docs/community/developer_guide.mdx [hunk 4]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
...
-1. Open a release PR to
-   - Build with latest golang: replace go version of [binary](https://github.com/oras-project/oras/blob/main/.github/workflows/release-github.yml#L32) and [image](https://github.com/oras-project/oras/blob/main/Dockerfile#L14) to latest stable one
-   - Update oras version: replace [current stable version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
-1. After the release PR got merged, [create an issue](https://github.com/oras-project/oras/issues/new) to call for vote on cutting off a release branch named `release-<major>.<minor>` based on the version update commit.
-1. Make fresh clone of the repo after all above steps are completed. Create a new tag for the version prefixed with "v" and push the tag directly to the repo.
+1. Open a PR to update Github Actions with latest version of golang: replace go version of [binary](https://github.com/oras-project/oras/blob/main/.github/workflows/release-github.yml#L32) and [image](https://github.com/oras-project/oras/blob/main/Dockerfile#L14) to latest stable one
+
+### Call for vote
```
Comment (FeynmanZhou) [inline docs/community/developer_guide.mdx:91]:
```suggestion
### Request to vote
```

File: docs/community/developer_guide.mdx [hunk 5]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
 This section needs a lot of love and automation 🙂
 
 :::
-
+### Preparation
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:86]: nit:
```suggestion
### Prerequisites
```

File: docs/community/developer_guide.mdx [hunk 6]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
...
-   - Update oras version: replace [current stable version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
-1. After the release PR got merged, [create an issue](https://github.com/oras-project/oras/issues/new) to call for vote on cutting off a release branch named `release-<major>.<minor>` based on the version update commit.
-1. Make fresh clone of the repo after all above steps are completed. Create a new tag for the version prefixed with "v" and push the tag directly to the repo.
+1. Open a PR to update Github Actions with latest version of golang: replace go version of [binary](https://github.com/oras-project/oras/blob/main/.github/workflows/release-github.yml#L32) and [image](https://github.com/oras-project/oras/blob/main/Dockerfile#L14) to latest stable one
+
+### Request to vote
+1. Open a release PR to update oras version: replace [current stable version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:92]: RC versions are not stable versions
```suggestion
1. Open a release PR to update oras version: replace [current version](https://github.com/oras-project/oras/blob/main/internal/version/version.go#L20) with upcoming release version
```

File: docs/community/developer_guide.mdx [hunk 7]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
...
+ - The target branch is `main`
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:93]: If we want to release `v1.1.0` or `v1.1.0-rc.2`, the target branch is `release-1.1` as we already have a release branch.

File: docs/community/developer_guide.mdx [hunk 8]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
...
+ - The title should be `bump: tag and release ORAS CLI v<major>.<minor>[-<pre-release>]`
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:94]: missing `<patch>`

File: docs/community/developer_guide.mdx [hunk 9]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
...
+ - The description must reference the digest of version bump up commit for voting
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:95]: nit: code style: consistency: 4 spaces for indentation

File: docs/community/developer_guide.mdx [hunk 10]
```diff
@@ -83,14 +83,23 @@ To update or add new dependencies, run `go get <package name>`.
...
+1. Send a message to slack channel to call for vote on releasing. If the vote passes, merge the PR.
+ - The target commit should be the SHA digest of the last commit in the release PR
+ - Make sure that the PR is merged with `Create a merge commit` option
+ - (optional) cut off a release branch named `release-<major>.<minor>` if there is not one already
```
Comment (shizhMSFT) [inline docs/community/developer_guide.mdx:99]:
```suggestion
 - (optional) cut off a release branch named `release-<major>.<minor>` if not present

**Comment by TerryHowe** [comment]
Conflicts and comments here

**Pull request closed without merging**